docs/source/topics/data_types.rst

```diff
@@ -492,8 +495,92 @@ Example of creating an ``PortableCode``:
         filepath_executable='executable.exe'
     )
 
+.. _topics:data_types:core:code:containerized:
+
+ContainerizedCode
+-----------------
+
+.. versionadded:: 2.1
+
+The :class:`~aiida.orm.nodes.data.code.containerized.ContainerizedCode` class allows running an executable within a container image on a target computer.
+The data plugin stores the following information in the database:
+
+* ``image_name``: The name of the container image (e.g., ``docker://alpine:3``).
+* ``filepath_executable``: The filepath of the executable within the container (e.g. ``/usr/bin/bash``).
+* ``engine_command``: The bash command to invoke the container image (e.g. ``singularity exec --bind $PWD:$PWD {image_name}``).
+  The exact form of this command will depend on the containerization technology that is used.
+* ``computer``: The :class:`~aiida.orm.computers.Computer` on which to run the container.
+
+.. important::
+
+    If the ``engine_command`` contains variables (such as in the ``singularity exec --bind $PWD:$PWD {image_name}`` example), it is crucial that the ``Computer`` needs to have the ``use_double_quotes`` setting set to ``True``.
+    By default, a ``Computer`` will use single quotes to escape command line arguments and so the ``$PWD`` would not be expanded.
+    If this wasn't defined as such when the ``Computer`` was created, it can be changed through the API:
+
+    .. code-block::
+
+        computer = load_computer('some-computer')
+        computer.set_use_double_quotes(True)
+
+.. note::
+
+    The container image can be already download and stored on the target computer or can reference an image on a container index, such as Docker Hub.
+    In the latter case, the first time that the containerized code is run, the image will be automatically pulled from the index, which may take a while if the image is large.
+
+
+.. _topics:data_types:core:code:installed:containerized:setup:
+
+Setup
+^^^^^
+
+A ``ContainerizedCode`` can be created through the CLI as well as the API.
+The following examples show how to setup running ``bash`` in a base Docker container through Singularity to be run on the ``Computer`` named ``some-computer``:
+
+.. tab-set::
+
+    .. tab-item:: CLI
+
+        .. code-block:: console
+
+            verdi code create core.code.containerized \
+                --non-interactive \
+                --label containerized-code \
+                --default-calc-job-plugin core.arithmetic.add \
+                --computer some-computer \
+                --filepath-executable "/bin/sh" \
+                --image-name "docker://alpine:3" \
+                --engine-command "singularity exec --bind $PWD:$PWD {image_name}"
```

Comment:
By the way, it appears to me that the file path in the `engine_command` is not supposed to be absolute, while we require the path to the executable of the `Code` to be absolute.

This is fine with me, just wanted to point it out.
I guess one can argue that the result of the calculation is going to be less dependent on the exact version of the container runtime than on the exact version of the simulation code.
 
P.S. There have been several instances in the past where I've found the requirement of an absolute file path to be onerous (e.g. I rebuilt packages with spack, the `module load` command would have remained the same but the absolute path changed because of some hash, so I had to re-setup all the codes). It is not _bad_ to ask for the absolute path of the Code, it's just a bit pedantic not to give the user the choice. 


P.P.S. Inside a container, the requirement of an absolute file path for the code executable arguably loses its original purpose since the provenance on the code version is carried by the image name. To keep things simple, we can keep the requirement nevertheless.

Reply:
> By the way, it appears to me that the file path in the engine_command is not supposed to be absolute, while we require the path to the executable of the Code to be absolute.

I actually already addressed this in the latest change. The `ContainerizedCode` overrides the `filepath_executable` property and no longer requires that it is absolute.

Reply:
Ah, cool!